lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -19,10 +20,11 @@ def initialize(serializer, options = {})
 
         def serializable_hash(options = nil)
           options ||= {}
+
           if serializer.respond_to?(:each)
-            serializable_hash_for_collection(serializer, options)
```

Comment:
Oh I see. I'm all up for that, although I don't think it falls within the scope of this PR. Moreover, I think we should have `serializable_hash` defined in `Base`, and have serializers override `serializable_hash_for_(collection|single_resource)`.